spec/jobs/digital_object_import_processing_job_spec.rb

```diff
@@ -150,21 +156,53 @@
       {}
     end
 
+    let(:digital_object_data_without_digital_object_type) do
+      {
+        'assign_uid' => 'new_uid'
+      }
+    end
+
+    let(:digital_object_data_with_conflicting_params) do
+      {
+        'uid' => digital_object.uid,
+        'assign_uid' => digital_object.uid,
+        'digital_object_type' => 'item'
+      }
+    end
+
     it 'finds an existing digital object when a uid is provided in the digital_object_data' do
       expect(
         described_class.digital_object_for_digital_object_data(existing_digital_object_data).new_record?
       ).to eq(false)
     end
 
-    it 'instantiates a new digital object of the expected type when a uid is NOT provided in the digital_object_data' do
-      obj = described_class.digital_object_for_digital_object_data(new_digital_object_data)
-      expect(obj).to be_a(DigitalObject::Item)
-      expect(obj.new_record?).to eq(true)
+    it 'raises an error if assign_uid or uid are not provided in the digital_object_data' do
```

Comment:
Related to my earlier comment, creating a new import with no `assign_uid` or `uid` actually is a valid scenario (for automatic uid generation by the system).